CMakeLists.txt

```diff
@@ -34,6 +34,9 @@ option(BUILD_SHARED_LIBS "Build shared libraries instead of static ones" ON)
 # option for using Python
 option(FF_USE_PYTHON "Enable Python" ON)
 
+# option to download pre-compiled NCCL/Legion libraries
+option(FF_USE_PRECOMPILED_LIBRARIES "Enable use of pre-compiled NCCL & Legion libraries, if available" ON)
+
```

Comment:
suuuuper nit pick: Might be good to use a different variable for each dependency as precompiled or not